### PR TITLE
NovAtel OEM6 Service

### PR DESCRIPTION
**Pull request by CatGarab**
Creating a Kubos service for communicating with a NovAtel OEM6 GPS receiver.

We're learning that service PRs tend to get a bit large, so this work will be broken up into several smaller PRs instead. Those smaller PRs will get merged into this branch. Once all the work has been completed, this PR will be merged into `master`.

Note: This PR currently includes some changes to `kubos-service`. These changes have been made in #63, but that hasn't been merged yet, so I have temporarily duplicated the work.

Phase 0 (what's initially in this PR) - Service Framework:
- Initial `Subsystem` structure
- Stubs for all queries and mutations which will be implemented
- `ping` query to verify communication with the service (but not the end device)
- `ack` query containing the last mutation executed
- `errors` query and mutation which will return the contents of the master error RefCell (note: nothing currently adds anything to it, so the contents will always be empty)
- Initial unit tests

Phase 1 - Things which don't require RX:
- [x] Passthrough
- [x] Hardware configuration (LOG command)

Phase 2 - Simple RX:
- [x] Read thread
- [x] No-op
- [x] Hardware test
- [x] Hardware test results (same as hardware test, just called as a query)

Phase 3 - Position Data:
- [x] Lock status
- [x] Lock info

Phase 4 - Things that I'm not entirely sure how to implement off-hand... :
- [x] Power (need to figure out what to do about uptime field)
- [x] System status

Phase 5 - Telemetry:
- [x] Nominal (power, system status, lock status, lock info)
- [x] Debug (config? TBD...)

### CODE REVIEW

**Comment by CatGarab** [comment]
[Novatel OEM GPS Service](https://trello.com/c/OdZ48nYr/315-novatel-oem-gps-service)

File: services/novatel-oem6-service/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+#![deny(missing_docs)]
+#![deny(warnings)]
+
+//! Kubos Service for interacting with a [NovAtel OEM6 High Precision GNSS Receiver](https://www.novatel.com/products/gnss-receivers/oem-receiver-boards/oem6-receivers/)
+//!
+//! # Configuration
+//!
+//! The service can be configured in the `/home/system/etc/config.toml` with the following fields:
+//! ```
+//! [novatel-oem6-service]
```
Comment (plauche) [inline services/novatel-oem6-service/src/main.rs]: (I missed this on the mai service docs) The ip/port configuration actually goes into an `addr` section:
```toml
[novatel-oem6-service.addr]
ip = "127.0.0.1"
port = 8082

File: services/novatel-oem6-service/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+mod schema;
+#[cfg(test)]
+mod tests;
+
+use kubos_service::{Config, Service};
+use model::Subsystem;
+use novatel_oem6_api::OEMResult;
+use schema::{MutationRoot, QueryRoot};
+
+// TODO: CHANGE THE UART BUS TO UART4!!!
```
Comment (plauche) [inline services/novatel-oem6-service/src/main.rs]: Or get the uart port from the config file :)

**Comment by CatGarab** [comment]
It's finally time! The behemoth is done!
It's possible there are some bugs when running on actual hardware, but those can be discovered and fixed in a new PR and this stuff can all just get merged

**Pull request merged**